### ISSUE TITLE
FileType detection while uploading is based upon first dot?

### ISSUE THREAD

**Issue by Wardormeur** [crowdin/crowdin-cli-2#35]
Hi,
We tried to migrate from the ruby CLI to the java one, and stubbled upon a potential falsy filetype detection 
Scenario: 

> Setting up crowdin (2.0-13) ...
File 'messages.po'-\| - OK
File 'users-auth-create-reset.text.txt'/- - ERROR
code : 7
message : Invalid file extension. Allowed are: *.txt

More details : https://circleci.com/gh/CoderDojo/cp-zen-platform/1025
Crowdin.yaml : https://github.com/CoderDojo/cp-translations/blob/master/crowdin.yaml
cc @danielBrierton

**Comment by Andrulko** [comment]
The configuration itself seems to be ok:

```
  -
    source: '/email-templates/auth-create-reset-en_US/users-auth-create-reset.text.ejs'
    dest: '/users-auth-create-reset.text.txt'
    translation: '/email-templates/auth-create-reset-%locale_with_underscore%/%file_name%.ejs'
    type: 'txt'
```

Checking right away

**Comment by Andrulko** [comment]
@Wardormeur @danielBrierton

Please add the following parameter on the top of your configuration:
preserve_hierarchy: true

**Comment by ihor-popyk** [comment]
Hello @Wardormeur, @DanielBrierton!
This problem will be fixed in Crowdin CLI 2.0.15  (#41)
Thanks for report!
